internal/ir/xds.go

```diff
@@ -1651,3 +1652,10 @@ type HTTPTimeout struct {
 	// The maximum duration of an HTTP connection.
 	MaxConnectionDuration *metav1.Duration `json:"maxConnectionDuration,omitempty" yaml:"maxConnectionDuration,omitempty"`
 }
+
+// TLSBundle contains hostname and ca file in []byte format.
+// +k8s:deepcopy-gen=true
+type TLSBundle struct {
```

Comment:
I'd suggest renaming https://github.com/envoyproxy/gateway/blob/cf46fbe776918ad19444e26d637ffcc79676ca23/internal/ir/xds.go#L275 to `TLSDownstreamConfig` and renaming this struct to `TLSUpstreamConfig` with 2 fields `SNI` & `CACertificate`

Reply:
changed the TLSBundle accordingly .
btw, i am not sure how to calculate the sni in the envoy cluster config. there is only one sni we can configure. and if we have multiple destinations and multiple sni how do you suggest handle this ? use auto_sni ?